src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java

```diff
@@ -87,73 +78,75 @@ public BasicAccessioningService(AccessionGenerator<MODEL, ACCESSION> accessionGe
      * @return
      */
     private Map<HASH, MODEL> mapHashOfMessages(List<? extends MODEL> messages) {
-        return messages.stream()
-                .collect(Collectors.toMap(summaryFunction.andThen(hashingFunction), e -> e, (r, o) -> r));
-    }
-
-    private Map<HASH, MODEL> filterNotExistingAccessions(Map<HASH, MODEL> hashToMessages,
-                                                         Map<HASH, ACCESSION> existingAccessions) {
-        return hashToMessages.entrySet().stream()
-                .filter(entry -> !existingAccessions.containsKey(entry.getKey()))
-                .collect(Collectors.toMap(entry -> entry.getKey(), entry -> entry.getValue()));
-    }
-
-    private Map<ACCESSION, MODEL> joinExistingAccessionsWithMessages(Map<HASH, ACCESSION> existingAccessions,
-                                                                     Map<HASH, MODEL> hashToMessages) {
-        return existingAccessions.entrySet().stream()
-                .collect(Collectors.toMap(e -> e.getValue(), e -> hashToMessages.get(e.getKey())));
+        return messages.stream().collect(Collectors.toMap(hashingFunction, e -> e, (r, o) -> r));
     }
 
     /**
-     * Execute {@link BasicAccessioningServiceSaveDelegate#doSaveAccessions(List)} This operation will generate two
+     * Execute {@link BasicAccessioningServiceSaveDelegate#doSaveAccessionedModels(List)} This operation will generate two
      * lists on {@link SaveResponse} saved elements and not saved elements. Not saved elements are elements that
      * could not be stored on database due to constraint exceptions. This should only happen when elements have been
      * already stored by another application instance / thread with a different id.
-     * See {@link #getUnsavedAccessions(SaveResponse)} for more details.
+     * See {@link #getUnsavedAccessions(List)} } for more details.
      *
      * @param accessions
      * @return
      */
-    private Map<ACCESSION, MODEL> saveAccessions(List<ModelHashAccession<MODEL, HASH, ACCESSION>> accessions) {
-        SaveResponse<ACCESSION, MODEL> response = basicAccessioningServiceSaveDelegate.doSaveAccessions(accessions);
+    private List<AccessionModel<MODEL, HASH, ACCESSION>> saveAccessions(List<AccessionModel<MODEL, HASH, ACCESSION>> accessions) {
+        SaveResponse<ACCESSION> response = basicAccessioningServiceSaveDelegate.doSaveAccessionedModels(accessions);
```

Comment:
Maybe this was a result of automated renaming, but I think the previous name was better: shorter and it gave the same information.